pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +391,26 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) []map[string]interface{} {
+	registries := make([]map[string]interface{}, len(registryMap))
+	for k, v := range registryMap {
+		registries[k] = v
+	}
+	return registries
+}
+
+func getIndexFromQuestion(question string) (int, error) {
+	var index int
+	var err error
+	re1, _ := regexp.Compile(`\[\d+\]$`)
+	target := re1.FindString(question)
+	if target != "" {
+		target = target[1 : len(target)-1]
+		index, err = strconv.Atoi(target)
```

Comment:
you could remove this error check, the regex will fail if it is not a valid number.  

NIT: this function is a good candidate for some unit tests.

Reply:
fixed bu removing the error check